UnityProject/Assets/Scripts/Cooldowns/HasCooldowns.cs

```diff
@@ -0,0 +1,101 @@
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// NOTE: You can use this (on a given object) or Cooldowns (static methods) as the main API for working with cooldowns.
+///
+/// Allows a game object to have cooldowns which can be triggered on, take a defined amount of time to
+/// go off, and can be checked if they are on/off.
+/// Intended to be used for a wide variety of things, such as interaction logic and abilities. A given cooldown
+/// can be used across multiple components as well, such as for various melee-related actions that should not
+/// be possible to spam all together.
+///
+/// Cooldowns are primarily defined using Cooldown assets, but for convenience there are also ways to use cooldowns
+/// for one-off situations without needing to create an asset.
+///
+/// At the moment, there is currently NO networking logic for cooldowns. The clients (incl. host player) and server track cooldowns totally
```

Comment:
Do you think there could be networking issues when a player joins mid-cooldown?

Reply:
I think the last part of the comment is sufficient enough "We can add networking logic for future cooldown use cases if it becomes necessary."

Most (if not all) cooldowns would be less then a second so it won't affect new players.